open_feature/open_feature_client.py

```diff
@@ -190,7 +258,11 @@ def evaluate_flag_details(
             client_metadata=None,
             provider_metadata=None,
         )
-        merged_hooks = self.hooks
+        merged_hooks = (
```

Comment:
It seems to me that the hooks should be in different orders depending on the stage in the evaluation lifecycle, whereas I think we're just defining a single order and passing these to each `x_hooks` function call?

https://github.com/open-feature/spec/blob/main/specification/sections/04-hooks.md#requirement-442

Specifically, it looks like before_hooks are evaluated in the opposite order to the rest of the hook evaluations.

Reply:
I've put in a new order for the other hooks